src/main/java/core/basesyntax/Pair.java

```diff
@@ -1,5 +1,43 @@
 package core.basesyntax;
 
-class Pair {
+import java.util.Objects;
 
+class Pair<T1, T2> {
+    private T1 first;
+    private T2 second;
+
+    private Pair(T1 first, T2 second) {
+        this.first = first;
+        this.second = second;
+    }
+
+    public static <T1, T2> Pair<T1, T2> of(T1 t1, T2 t2) {
+        return new Pair<>(t1, t2);
+    }
+
+    T1 getFirst() {
+        return this.first;
+    }
+
+    T2 getSecond() {
+        return this.second;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Pair<?, ?> pair = (Pair<?, ?>) o;
+        return Objects.equals(first, pair.first)
+                && Objects.equals(second, pair.second);
```

Comment:
try to solve this task without using Objects class, same in hashcode() 
it is mentioned in checklist, please check your solution with it before sending for review